frontend/appflowy_flutter/lib/workspace/presentation/home/panes/flowy_pane_group.dart

```diff
@@ -0,0 +1,141 @@
+import 'package:appflowy/workspace/application/panes/panes.dart';
+import 'package:appflowy/workspace/application/panes/panes_cubit/panes_cubit.dart';
+import 'package:appflowy/workspace/application/panes/size_controller.dart';
+import 'package:appflowy/workspace/presentation/home/home_layout.dart';
+import 'package:appflowy/workspace/presentation/home/home_stack.dart';
+import 'package:appflowy/workspace/presentation/home/panes/panes_layout.dart';
+import 'package:flutter/gestures.dart';
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+
+import 'flowy_pane.dart';
+
+class FlowyPaneGroup extends StatelessWidget {
+  final PaneNode node;
+  final double groupWidth;
+  final double groupHeight;
+  final HomeLayout layout;
+  final HomeStackDelegate delegate;
+  const FlowyPaneGroup({
+    super.key,
+    required this.node,
+    required this.groupWidth,
+    required this.groupHeight,
+    required this.layout,
+    required this.delegate,
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    if (node.children.isEmpty) {
+      return FlowyPane(
+        node: node,
+        delegate: delegate,
+        layout: layout,
+      );
+    } else {
+      return SizedBox(
+        child: LayoutBuilder(
+          builder: (context, constraints) => Stack(
+            key: ValueKey(node.paneId),
+            children: [
+              ...node.children.indexed
+                  .map(
+                    (indexNode) => GestureDetector(
+                      child: ChangeNotifierProvider<PaneSizeController>(
+                        create: (context) => node.sizeController,
+                        builder: (context, widget) =>
```

Comment:
Is `builder` required? Couldn't you just use `child` here? _(I'm not too into change notifier providers, just curious)_

Reply:
Same here, not too into change notifiers either, I am hoping we will be able to refactor all these to blocs if we persist some data, will look into it though.